commcare_connect/opportunity/tasks.py

```diff
@@ -286,3 +286,22 @@ def approve_completed_work_and_update_payment_accrued(completed_work_ids: list[i
                 access.payment_accrued += completed_count * completed_work.payment_unit.amount
                 completed_work.save()
                 access.save()
+
+
+@celery_app.task()
+def bulk_approve_completed_work():
+    access_objects = OpportunityAccess.objects.filter(
+        opportunity__active=True,
+        opportunity__end_date__gte=datetime.date.today(),
+    )
+    for access in access_objects:
+        completed_works = access.completedwork_set.filter(status=CompletedWorkStatus.pending)
+        for completed_work in completed_works:
```

Comment:
shouldn't this check if the setting to auto approve is enabled for that opportunity?

Reply:
Fixed. https://github.com/dimagi/commcare-connect/pull/294/commits/7635284b54b8c16e0bd12415bdc2ce54997fd4d9